File: LeetCode/0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int sum = 0, n = nums.size(), MAX = 0;
        for (int i = 0; i < k; i++)
            sum += nums[i];
        MAX = sum;
        for (int i = 1; i <= n - k; i++) {
            sum -= nums[i - 1];
            sum += nums[i + k - 1];
            MAX = max(sum, MAX);
        }
        return (double) MAX / k;
    }
};

File: LeetCode/0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        slide = sum(nums[:k])
        max_sum = slide

        for i in range(k, len(nums)):
            slide += -nums[i - k] + nums[i]
            max_sum = max(max_sum, slide)

        return max_sum / k
